libsnark/zk_proof_systems/plonk/srs.tcc

```diff
@@ -56,38 +54,194 @@ srs<ppT>::srs(
 {
 }
 
-/// class plonk_verification_key
 template<typename ppT>
 plonk_verification_key<ppT>::plonk_verification_key(
     std::vector<libff::G2<ppT>> &&secret_powers_g2)
     : secret_powers_g2(std::move(secret_powers_g2)){};
 
-/// class plonk_keypair constructor
 template<typename ppT>
 plonk_keypair<ppT>::plonk_keypair(
     plonk_proving_key<ppT> &&pk, plonk_verification_key<ppT> &&vk)
     : pk(std::move(pk)), vk(std::move(vk))
 {
 }
 
-/// struct transcript_hash_t constructor
+template<typename ppT> transcript_hasher<ppT>::transcript_hasher()
```

Comment:
This is basically a test vector. Ideally the main lib would define an interface and the implementation that holds these test vectors would be part of the tests dir. Further, if these test vector valves are only valid for BLS 12-381 then this test implementation wouldonly exist for that curve, not for all ppT.

Reply:
Could potentially of addressed later if you prefer to create an issue.

Reply:
Indeed these test vector values are already part of the class `plonk_example` in `tests/example.*` . I created an object of this class inside `transcript_hasher` and extracted the values from the object. 

I would like to discuss the second part i.e. making the test values available only for `BLS12-381`, before proceeding with the change.

Reply:
> if these test vector values are only valid for BLS 12-381 then this test implementation wouldonly exist for that curve, not for all ppT.

Is it true that this and also [this](https://github.com/clearmatics/libsnark/pull/61#discussion_r964178468) is (i.e. will be) addressed by implementing the proposed steps 1-4 https://github.com/clearmatics/libsnark/pull/61#discussion_r966911185 ?

Reply:
Yes, that would address it.